### ISSUE TITLE
cos and sin for add_missing_variable

### ISSUE THREAD

**Issue by MaceKuailv** [malmans2/oceanspy#234]
I wrote a small function that uses spherical triangles to approximate the grid orientation variables (CS and SN) based on lat-lon. It should be handy as a part of add_missing_variables

The core function looks something like this. 

```
def find_cs_sn(thetaA,phiA,thetaB,phiB):
    """
    theta is the angle 
    between the meridian crossing point A
    and the geodesic connecting A and B
    
    this function return cos and sin of theta
    """
    # O being north pole
    AO = np.pi/2 - thetaA
    BO = np.pi/2 - thetaB
    dphi = phiB-phiA
    # Spherical law of cosine on AOB
    cos_AB = np.cos(BO)*np.cos(AO)+np.sin(BO)*np.sin(AO)*np.cos(dphi)
    sin_AB = np.sqrt(1-cos_AB**2)
    # spherical law of sine on triangle AOB
    SN = np.sin(BO)*np.sin(dphi)/sin_AB
    CS = np.sign(thetaB-thetaA)*np.sqrt(1-SN**2)
    return CS,SN
```

P.S. I actually wrote a similar code a few months ago, but I lost the code and had to do it again.

**Comment by ThomasHaine** [comment]
Thanks @MaceKuailv!  How do you envisage this function being used?

**Comment by MaceKuailv** [comment]
It will go into the compute.py module. This function will be called when one uses od._add_missing_variables and 'CS' and/or 'SN' is in the varList argument.

**Comment by Mikejmnez** [comment]
Actually, I didn't realize that the variables `CS` and `SN` where going to be used so I dropped them from the dataset. You can see from the catalog in `/oceanspy/sciserver_catalogs/catalog_xarray.yaml` in line 726 when reading the grid. Easy fix.

I am working on a PR that will also fix issue #223 . I will include it there.

**Comment by Mikejmnez** [comment]
closed by PR #244
